### ISSUE TITLE
ExtAuthz with HTTP server does not send raw bytes in request body with a multipart/form-data

### ISSUE THREAD

**Issue by spudfkc** [envoyproxy/envoy#27220]
*Title*: ExtAuthz with HTTP server does not send raw bytes in request body with a multipart/form-data 

*Description*:
I have an ExtAuthz configuration for an HTTP auth server and configured `with_request_body`. However, when I set `pack_as_bytes=true`, the forwarded request has no HTTP body. The incoming requests have a request body that is serialized to bytes, so the body should not be sanitized. 

I would expect that the body of the HTTP request going to the auth service would be the raw bytes from the originating request. But there is no request body. The documentation says that when `pack_as_bytes=true` the raw body will be in the `raw_body` field, but it's not clear how that works with an HTTP server. 

However, if I set `pack_as_bytes=false`, the request body is there, but as of envoy version 1.26.0, the request body gets sanitized as a UTF8 string. I am able to work around this by setting `envoy.reloadable_features.service_sanitize_non_utf8_strings=false` but ideally I would like to use `pack_as_bytes=true`.

*Repro steps*:
1. Use the provided Envoy configuration
2. Have an HTTP service available at `http://extauth.test.svc.cluster.local:80` 
3. Make an HTTP request to Envoy. The request body should be bytes that include invalid UTF8 bytes. E.x. `curl -v -XPOST 'http://envoy/foo' --form 'message=@"~/somefile.bin"'` An example file be populated with a byte-serialized message with something like [msgpack](https://msgpack.org/) (which is what I'm using in my scenario).
4. The ExtAuthz server will receive an HTTP request with an empty request body.

>**Note**: The [Envoy_collect tool](https://github.com/envoyproxy/envoy/blob/main/tools/envoy_collect/README.md)
gathers a tarball with debug logs, config and the following admin
endpoints: /stats, /clusters and /server_info. Please note if there are
privacy concerns, sanitize the data prior to sharing the tarball/pasting.

*Admin and Stats Output*:
>Include the admin output for the following endpoints: /stats,
/clusters, /routes, /server_info. For more information, refer to the
[admin endpoint documentation.](https://www.envoyproxy.io/docs/envoy/latest/operations/admin)

>**Note**: If there are privacy concerns, sanitize the data prior to
sharing.
N/A

*Config*:
Example config (I am running this via an EnvoyFilter resource in a k8s cluster deployed with istio)
```yaml
name: envoy.filters.http.ext_authz.test
typedConfig:
  '@type': type.googleapis.com/envoy.extensions.filters.http.ext_authz.v3.ExtAuthz
  withRequestBody:
    pack_as_bytes: true
    allow_partial_message: true
    max_request_bytes: 1024
  httpService:
    authorizationRequest:
      headersToAdd:
        - key: "x-forwarded-host"
          value: "%REQ(host)%"
        - key: "accept"
          value: "application/x-protobuf"
      allowedHeaders:
        patterns:
          - exact: id1
            ignore_case: true
          - exact: id2
            ignore_case: true
          - exact: content-type
            ignore_case: true
    authorizationResponse:
      allowed_upstream_headers:
        patterns:
          - exact: device-context
            ignore_case: true
    pathPrefix: /v1/auth
    serverUri:
      cluster: outbound|80||extauth.test.svc.cluster.local
      timeout: 5s
      uri: http://extauth.test.svc.cluster.local:80
  statusOnError:
    code: 503
```

*Logs*:
>Include the access logs and the Envoy logs.
N/A

>**Note**: If there are privacy concerns, sanitize the data prior to
sharing.
N/A

*Call Stack*:
> If the Envoy binary is crashing, a call stack is **required**.
Please refer to the [Bazel Stack trace documentation](https://github.com/envoyproxy/envoy/tree/main/bazel#stack-trace-symbol-resolution).
N/A

**Comment by wbpcode** [comment]
After a quick check, it's bug of ext_authz. For the http client, it will always get request body from original `body` field.
A simple patch is necessary.

**Comment by tejagade1** [comment]
When will this fix be released? We are seeing this issue as well and looking at the release schedule [here](https://github.com/envoyproxy/envoy/blob/main/RELEASES.md), it looks like `1.27.0` won't be released until July. Can we have a hotfix for this bug?

**Comment by wbpcode** [comment]
It all you are want this to be backported, then I can backport this patch to the 1.26. But I also don't know when will 1.26 will cust a security release.

**Comment by tejagade1** [comment]
Yes, patching it on 1.26 works for us.

**Comment by tejagade1** [comment]
@wbpcode Thank you for merging this to the 1.26 release branch. Do you know when can we expect this to be released on 1.26?

**Comment by wbpcode** [comment]
I think it might be released recently. The @phlax is preparing some patchs.
